ocaml/xapi/livepatch.ml

```diff
@@ -0,0 +1,279 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module D = Debug.Make (struct let name = "livepatch" end)
+
+open D
+
+exception Invalid_component
+
+type component = Xen | Kernel
+
+let component_of_string x =
+  match Astring.String.Ascii.lowercase x with
+  | "xen" ->
+      Xen
+  | "kernel" ->
+      Kernel
+  | _ ->
+      raise Invalid_component
+
+let pkg_name_of_component = function
+  | Xen ->
+      "xen-hypervisor"
+  | Kernel ->
+      "kernel"
+
+let string_of_component = function Xen -> "xen" | Kernel -> "kernel"
+
+type t = {
+    component: component
+  ; base_build_id: string
+  ; to_version: string option
```

Comment:
I am bit surprised about `to_` - it is somewhat unusual. Does this indicate the version and release reached after applying the patch? Maybe a comment could clarify this.

Reply:
Yes. It indicates that the version and release reached after applying the live patch.

Reply:
I see that there are comments that explain this in the mli file.